arangod/Replication2/Supervision/CollectionGroupSupervision.cpp

```diff
@@ -601,6 +603,27 @@ struct TransactionBuilder {
               .end();
   }
 
+  void operator()(UpdateCollectionPlan const& action) {
+    auto tmp = env.write();
+    auto allProps = velocypack::serialize(action.spec);
+    ADB_PROD_ASSERT(allProps.isObject());
+    for (auto const& it : VPackObjectIterator(allProps.slice())) {
+      tmp = std::move(tmp).emplace_object(
+          basics::StringUtils::concatT("/arango/Plan/Collections/", database,
+                                       "/", action.cid, "/",
+                                       it.key.stringView()),
+          [&](VPackBuilder& builder) {
+            velocypack::serialize(builder, it.value);
```

Comment:
Question: if the value is null, does that write a `null` into the agency, or removes the property? Do we care about it, or it's not a problem?

Reply:
That is a good spot!

If the value "null" is added for schema, then schema will not be serialized (as null here is equal to non-existent with the std::optional)
This way the update actually does not remove the schema (or set to null) with this method.
Will fix

Reply:
Fixed: a6260d57ff6